waterlog/src/actions/CostsReportActions.js

```diff
@@ -0,0 +1,37 @@
+import {
+  FETCH_COSTS_DAILY,
+  FETCH_COSTS_MONTHLY,
+  FETCH_COSTS_SEASONALLY
+} from './Types';
+import { Globals } from './../Globals';
+
+export const fetchCostsDaily = () => (dispatch) => {
+  fetch(`${Globals.API_URL}/api/segmentevents/dailyCost`) //Change to use either localhost/server
+    .then(res => res.json())
+    .then(dailyCost => {
+      dispatch({
+        type: FETCH_COSTS_DAILY,
+        payload: dailyCost
+      });
+    });
+};
+export const fetchCostsMonthly = () => (dispatch) => {
+  fetch(`${Globals.API_URL}/api/segmentevents/monthlyCost`) //Change to use either localhost/server
+    .then(res => res.json())
+    .then(monthlyCost => {
+      dispatch({
+        type: FETCH_COSTS_MONTHLY,
+        payload: monthlyCost
+      });
+    });
+};
+export const fetchCostsSeasonally = () => (dispatch) => {
```

Comment:
These 3 functions are quite the same except for the endpoints, the type and the payload. Don't you think you can make a function and just pass those in? Or even better, remove all functions and make a general one e.g. fetchCosts(Endpoints.seasonallyCost, FETCH_COSTS_SEASONALLY, seasonalCost)

Reply:
These functions like the reducers are also not complete, I am still going to update them to do error handling in the next PR.

Reply:
Ok